src/hooks/useGiftRanking.ts

```diff
@@ -0,0 +1,13 @@
+import useApiRequest from './useApiRequest';
+import { fetchRanking } from '@/api/RankingApi';
+import type { Product } from '@/types/Product';
+
+export default function useGiftRanking(gender: string, type: string) {
+  const fetchData = () => fetchRanking(gender, type);
+  const {
+    data: products,
+    isLoading,
+    hasError,
+  } = useApiRequest<Product[]>(fetchData, [gender, type]);
+  return { products, isLoading, hasError };
+}
```

Comment:
앞선 피드백을 토대로 이렇게 개선할 수 있답니다!

```suggestion
export default function useGiftRanking(gender: string, type: string) {
  return useApiRequest(fetchRanking, gender, type)
}
```